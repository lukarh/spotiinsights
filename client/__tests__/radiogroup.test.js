import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { act, render, cleanup, fireEvent } from "@testing-library/react";
import TimeRadioGroup from '../src/components/FunctionalComponents/TimeRadioGroup';

describe('Time Radio Group Component', () => {

    beforeEach(() => {
        jest.clearAllMocks()
    })

    afterEach(() => {
        cleanup()
    })

    mockHandleTimeChange = jest.fn()

    const currentLabel = 'Current'
    const prev6MonthsLabel = 'Last 6 Months'
    const allTimeLabel = 'All-Time'

    const currentValue = 'short_term'
    const prev6MonthsValue = 'medium_term'
    const allTimeValue = 'long_term'

    test('should render with correct option labels and default label selected', () => {
        const { getByLabelText } = render(<TimeRadioGroup handleTimeChange={mockHandleTimeChange}/>)

        const currentRadioOption = getByLabelText(currentLabel)
        const last6MonthsRadioOption = getByLabelText(prev6MonthsLabel)
        const allTimeRadioOption = getByLabelText(allTimeLabel)

        expect(currentRadioOption).toBeInTheDocument()
        expect(last6MonthsRadioOption).toBeInTheDocument()
        expect(allTimeRadioOption).toBeInTheDocument()

        expect(currentRadioOption.checked).toEqual(false)
        expect(last6MonthsRadioOption.checked).toEqual(true)
        expect(allTimeRadioOption.checked).toEqual(false)
    })

    test('should render options and onClick should change selected radio option', async () => {
        const { getByLabelText } = render(<TimeRadioGroup handleTimeChange={mockHandleTimeChange}/>)

        const currentRadioOption = getByLabelText(currentLabel)
        const last6MonthsRadioOption = getByLabelText(prev6MonthsLabel)
        const allTimeRadioOption = getByLabelText(allTimeLabel)

        await act(async () => {
            fireEvent.click(currentRadioOption)
        })

        fireEvent.click(currentRadioOption, { target: { checked: true } })

        expect(currentRadioOption.checked).toEqual(true)
        expect(last6MonthsRadioOption.checked).toEqual(false)
        expect(allTimeRadioOption.checked).toEqual(false)

        fireEvent.click(allTimeRadioOption, { target: { checked: true } })

        expect(currentRadioOption.checked).toEqual(false)
        expect(last6MonthsRadioOption.checked).toEqual(false)
        expect(allTimeRadioOption.checked).toEqual(true)
    })

    test('should handle callback function on every radio option click', async () => {
        const { getByLabelText } = render(<TimeRadioGroup handleTimeChange={mockHandleTimeChange}/>)

        const currentRadioOption = getByLabelText(currentLabel)
        const last6MonthsRadioOption = getByLabelText(prev6MonthsLabel)
        const allTimeRadioOption = getByLabelText(allTimeLabel)

        await act(async () => {
            fireEvent.click(currentRadioOption)
        })

        expect(mockHandleTimeChange).toHaveBeenCalledTimes(1)
        expect(mockHandleTimeChange).toHaveBeenCalledWith(
            expect.objectContaining({
                target: expect.objectContaining({ value: currentValue }),
            }),
            currentValue
        )

        await act(async () => {
            fireEvent.click(allTimeRadioOption)
        })

        expect(mockHandleTimeChange).toHaveBeenCalledTimes(2)
        expect(mockHandleTimeChange).toHaveBeenCalledWith(
            expect.objectContaining({
                target: expect.objectContaining({ value: allTimeValue }),
            }),
            allTimeValue
        )

    })

    test('should match snapshot', () => {
        const tree = renderer.create(
            <TimeRadioGroup handleTimeChange={mockHandleTimeChange} />
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })

})


// {"_reactName": "onChange", 
// "_targetInst": null, 
// "bubbles": true, 
// "cancelable": true, 
// "currentTarget": null, 
// "defaultPrevented": false, 
// "eventPhase": 3, 
// "isDefaultPrevented": [Function functionThatReturnsFalse],
//  "isPropagationStopped": [Function functionThatReturnsFalse], 
//  "isTrusted": false, "nativeEvent": {"isTrusted": false}, 
//  "target": <input class="PrivateSwitchBase-input css-1m9pwf3" name=":r2:" type="radio" value="long_term" />, 
//  "timeStamp": 1691190322276, 
//  "type": "change"}, 
//  "long_term"