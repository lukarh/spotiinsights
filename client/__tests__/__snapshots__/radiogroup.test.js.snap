// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Time Radio Group Component should match snapshot 1`] = `
<div
  className="MuiStack-root padding-1-rem css-1z0rh7t-MuiStack-root"
>
  <div>
    <h1>
      Find out your Top 10s for any of the following time periods:
    </h1>
  </div>
  <div
    className="radio-group-container padding-1-rem"
  >
    <div
      className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
      style={
        {
          "fontFamily": "DM Sans",
        }
      }
    >
      <div
        className="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
        role="radiogroup"
        style={
          {
            "display": "flex",
            "flexDirection": "row",
            "fontFamily": "DM Sans",
            "justifyContent": "space-between",
          }
        }
      >
        <label
          className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          style={
            {
              "fontFamily": "DM Sans",
            }
          }
        >
          <span
            className="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <input
              checked={false}
              className="PrivateSwitchBase-input css-1m9pwf3"
              disabled={false}
              name=":r0:"
              onChange={[Function]}
              required={false}
              type="radio"
              value="short_term"
            />
            <span
              className="css-hyxlzm"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                data-testid="RadioButtonUncheckedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                data-testid="RadioButtonCheckedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                />
              </svg>
            </span>
          </span>
          <span
            className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
          >
            Current
          </span>
        </label>
        <label
          className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          style={
            {
              "fontFamily": "DM Sans",
            }
          }
        >
          <span
            className="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary Mui-checked MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <input
              checked={true}
              className="PrivateSwitchBase-input css-1m9pwf3"
              disabled={false}
              name=":r0:"
              onChange={[Function]}
              required={false}
              type="radio"
              value="medium_term"
            />
            <span
              className="css-hyxlzm"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                data-testid="RadioButtonUncheckedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-11zohuh-MuiSvgIcon-root"
                data-testid="RadioButtonCheckedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                />
              </svg>
            </span>
          </span>
          <span
            className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
          >
            Last 6 Months
          </span>
        </label>
        <label
          className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          style={
            {
              "fontFamily": "DM Sans",
            }
          }
        >
          <span
            className="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <input
              checked={false}
              className="PrivateSwitchBase-input css-1m9pwf3"
              disabled={false}
              name=":r0:"
              onChange={[Function]}
              required={false}
              type="radio"
              value="long_term"
            />
            <span
              className="css-hyxlzm"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                data-testid="RadioButtonUncheckedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                data-testid="RadioButtonCheckedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                />
              </svg>
            </span>
          </span>
          <span
            className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
          >
            All-Time
          </span>
        </label>
      </div>
    </div>
  </div>
</div>
`;
